oak_runtime/src/lib.rs

```diff
@@ -575,9 +588,15 @@ impl Runtime {
     fn get_node_downgraded_label(&self, node_id: NodeId, initial_label: &Label) -> Label {
         // Retrieve the set of tags that the node may downgrade.
         let node_privilege = self.get_node_privilege(node_id);
-        Label {
-            // Remove all the confidentiality tags that the Node may declassify.
-            confidentiality_tags: initial_label
+
+        // Remove all the confidentiality tags that the Node may declassify.
```

Comment:
@aferr is there a more elegant way of computing a downgraded label in general, perhaps using meet and join operator from the underlying principal lattice, if we had them?

Reply:
If we go with the NMIFC rules for downgrading, meet/join would be closer to how its done in that paper and the related ones. With privilege as a separate thing from integrity, it might work out similarly if you just take the type rule from that paper and replace the integrity projection with the downgrade privilege. I haven't fully thought this out though. 

I think the way you have it here is perfectly fine, though.

Reply:
Do you have a link to the paper?